### PR TITLE
Fixes #15590: Renaming/moving technique category fails in several way

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15590

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/TechniqueRepository.scala [hunk 1]
```diff
@@ -122,14 +122,20 @@ trait TechniqueRepository {
...
   def getParentTechniqueCategory_forTechnique(id: TechniqueId): Box[TechniqueCategory]
 
-  final def getTechniqueCategoriesBreadCrump(id: TechniqueId): Box[Seq[TechniqueCategory]] = {
+  final
+  def getTechniqueCategoriesBreadCrump(id: TechniqueId): Box[Seq[TechniqueCategory]] = {
     for {
       cat <- getParentTechniqueCategory_forTechnique(id)
       path <- sequence(cat.id.getIdPathFromRoot) { currentCatId =>
-        getTechniqueCategory(currentCatId) ?~! "'%s' category was not found but should be a parent of '%s'".format(currentCatId, cat.id)
+        getTechniqueCategory(currentCatId) ?~! "'%s' category was not found but should be a parent of '%s'".format(
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/TechniqueRepository.scala]: You could have switch to a string interpolation ! and more importantly you should use currentCatId.,name and cat.id.name

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**